android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -169,65 +169,63 @@ protected void performUpdate() {
       }
     }
 
-    boolean customAnimation = false;
+    // variables used for "open/back animation" logic
+    boolean shouldPlayOpenAnimation = true;
+    boolean isCustomAnimation = false;
+    int transition;
+    Screen.StackAnimation stackAnimation = Screen.StackAnimation.DEFAULT;
 
     if (!mStack.contains(newTop)) {
       // if new top screen wasn't on stack we do "open animation" so long it is not the very first screen on stack
       if (mTopScreen != null && newTop != null) {
         // there was some other screen attached before
-        int transition = FragmentTransaction.TRANSIT_FRAGMENT_OPEN;
-        if (!mScreenFragments.contains(mTopScreen) && newTop.getScreen().getReplaceAnimation() == Screen.ReplaceAnimation.POP) {
-          // if the previous top screen does not exist anymore and the new top was not on the stack before,
-          // probably replace was called, so we check the animation
-          transition = FragmentTransaction.TRANSIT_FRAGMENT_CLOSE;
-        }
-        switch (newTop.getScreen().getStackAnimation()) {
-          case NONE:
-            transition = FragmentTransaction.TRANSIT_NONE;
-            break;
-          case FADE:
-            transition = FragmentTransaction.TRANSIT_FRAGMENT_FADE;
-            break;
-          case SLIDE_FROM_RIGHT:
-            customAnimation = true;
-            getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_right, R.anim.rns_slide_out_to_left);
-            break;
-          case SLIDE_FROM_LEFT:
-            customAnimation = true;
-            getOrCreateTransaction().setCustomAnimations(R.anim.rns_slide_in_from_left, R.anim.rns_slide_out_to_right);
-            break;
-        }
-
-        if (!customAnimation) {
-          getOrCreateTransaction().setTransition(transition);
-        }
+        // if the previous top screen does not exist anymore and the new top was not on the stack before,
+        // probably replace or reset was called, so we play the "back animation"
+        // otherwise it's open animation
+        shouldPlayOpenAnimation = mScreenFragments.contains(mTopScreen) || newTop.getScreen().getReplaceAnimation() != Screen.ReplaceAnimation.POP;
+        // and check if it is a custom animation
+        isCustomAnimation = isCustomAnimation(newTop);
```

Comment:
We check the `stackAnimation` the line below, so why not make this check use this value instead of retrieving the `Screen` etc. again? Also, I think the name of the method shouldn't be the same as the variable name. Maybe the variable name could be something like `shouldUseCustomAnimation`. Same for few lines below.

Reply:
Fixed in 7e8721b08db4ceda2942dc73f00470caff78edd8